.travis.yml

```diff
@@ -9,37 +9,56 @@ addons:
     packages:
       - docker-ce
       - jq
+# only push builds on master, deploy, release branches to prevent double build or
+# it's not push initiated
+# or it is forced by a commit message
+# but never build forks (can be dangerous to run foreign code, right?)
+if: (branch IN (master, deploy, release) OR type IN (pull_request, api, cron) OR commit_message =~ /#travis/) AND fork = false
 matrix:
   allow_failures:
     - python: 3.8-dev
+
+  fast_finish: true
   include:
+
     - stage: Run tests
       language: node_js
-      name: frontend E2E cypress tests
+      name: Compose test environment and run e2e tests
+
+    # run cypress tests
       node_js: 10
       cache:
         directories:
-          - node_modules
```

Comment:
Can you please explain why this was removed?

Reply:
caching node_modules caches the state of last app installation while we need a fresh environment to do tests. `.npm` directory, on the other hand, is just a cache.